src/main/java/io/vertx/proton/impl/ProtonConnectionImpl.java

```diff
@@ -371,12 +367,14 @@ public void fireDisconnect() {
     }
   }
 
-  void bindClient(NetClient client, NetSocket socket, ProtonSaslClientAuthenticatorImpl authenticator) {
+  void bindClient(NetClient client, NetSocket socket, ProtonSaslClientAuthenticatorImpl authenticator, ProtonClientOptions options) {
     transport = new ProtonTransport(connection, vertx, client, socket, authenticator);
+    transport.setIdleTimeout(options.getHeartbeat() * 2);
```

Comment:
I'd probably pass this via the constructor instead of as a setter, maybe even adding some transport options to avoid changing it for similar things later, nothing can or should change this value beyond the construction point.